Add ability to name markers

mwcraig opened the pull request:
This PR (which really needs an update to the example notebook before it is merged) adds the ability to name sets of markers. If the user does not specify a name, then a default name is used (i.e. the old behavior, essentially).

Markers added interactively, as the result of setting `ImageWidget.is_marking = True` and then clicking, are added to the display with a special name.

There are a couple user-facing changes:

+ `add_markers` now accepts a `marker_name` argument. If, for example, I was was adding all of the APASS stars in the image I might choose the name `'APASS'` for them.
+ `remove_markers` also accepts a `marker_name` argument. To remove my APASS markers I would can `remove_markers(marker_name='APASS')`. Any other markers I had added with a different name (or with no name) would remain.
+ `get_markers` accepts a `marker_name` argument. It also accepts a special value for the marker name, `'all'`, which returns locations for all of the markers.

Ping @eteq because I think this helps address #16. Interactive points are added with the marker name `interactive-markers`. 

If we want the user to choose the name for interactive markers, either because the user doesn't like this name or because they want to do several sets of interactive markers, then we should decide what the API is.

pllim commented on astrowidgets/core.py:
```diff
@@ -24,6 +24,9 @@
 ALLOWED_CURSOR_LOCATIONS = ['top', 'bottom', None]
 
+# A couple of marker names are for internal use only
+RESERVED_MARKER_SET_NAMES = ['all']
```
The comment says "a couple" but I only see one defined. I am confused.

pllim commented on astrowidgets/core.py:
```diff
@@ -112,6 +115,12 @@ def __init__(self, logger=None, image_width=500, image_height=500,
         # Marker
         self.marker = {'type': 'circle', 'color': 'cyan', 'radius': 20}
         self._marktag = 'marktag'
```
`self._marktag` is essentially useless now and hence should be removed accordingly from this entire module.

pllim commented on astrowidgets/core.py:
```diff
@@ -443,16 +455,28 @@ def get_markers(self, x_colname='x', y_colname='y',
         xy_col = []
 
-        if image.wcs.wcs is None:  # Do not include SkyCoord column
+        if (image is None) or (image.wcs.wcs is None):
```
Why is `image is None` comparison needed here? One cannot mark on an empty image widget, so at this point of logic, the image has to exist, no?

mwcraig replied:
In practice one *should* always have an image, but it is possible with the current API to add markers in the absence of an image. They don't display, since display only happens when there is an image. I think this came up when I was writing tests...

pllim commented on astrowidgets/core.py:
```diff
@@ -500,12 +524,13 @@ def get_markers(self, x_colname='x', y_colname='y',
                 [xy_col[:, 0], xy_col[:, 1], sky_col],
                 names=(x_colname, y_colname, skycoord_colname))
         else:
-            markers_table = Table(xy_col.T, names=(x_colname, y_colname))
```
Why is `.T` removed? Was it a bug? Has the API changed? I remember adding `.T` but I don't remember why.

mwcraig replied:
It was a bug...the `.T` raised an error.

pllim replied:
Weird... Pretty sure it worked when I tested it a while ago. Ah, well. :woman_shrugging:

pllim commented on astrowidgets/tests/test_image_widget.py:
```diff
@@ -0,0 +1,219 @@
+    data_y = 200
+    image._mouse_move_cb(image._viewer, None, data_x, data_y)
+    output_contents = image._jup_coord.value
+    print(output_contents)
```
Is this print necessary?

mwcraig replied:
nope, deleted

pllim commented on astrowidgets/tests/test_image_widget.py:
```diff
@@ -0,0 +1,219 @@
+    y_out = y_out.groups(1)[0]
+    assert float(x_out) == data_x + offset
+    assert float(y_out) == data_y + offset
+    # image.print_out.get_state()['outputs']
```
Remove commented code here.

mwcraig commented:
@pllim -- I think I've addressed the review comments and I've added a notebook demonstrating how marker names work. There are still some rough edges, e.g. some of the attributes I have as private should probably be public, like `ImageWidget._interactive_marker_set_name`
